File: uLab/Assets/Scripts/Test/ShopWindow.cs


using Locke;
using Locke.ui;
using UnityEngine;
using UnityEngine.UI;

public class ShopWindow : IWindow
{
	private static string[] buttonNames = new string[5] { "bag", "role", "shop", "tip", "dialog" };

	protected override void OnInit()
	{
		for (int i = 0; i < 5; i++)
		{
			Button button = this.FindWidget<Button>("buttons/button" + (i + 1));
			button.transform.GetComponentInChildren<Text>().text = buttonNames[i];
			UIEventListener.Get(button.gameObject).onClick = OnButtonClick;
		}

		Button back = FindWidget<Button>("buttons/back");
		back.onClick.AddListener(OnClick);
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

	private void OnButtonClick(GameObject go)
	{
		int index = int.Parse(go.name.Substring(go.name.Length - 1));

		if (index == 1)
		{
			UIManager.Instance.OpenWindow(GameUIList.bag);
		}
		else if (index == 2)
		{
			UIManager.Instance.OpenWindow(GameUIList.role);
		}
		else if (index == 3)
		{
			UIManager.Instance.OpenWindow(GameUIList.shop);
		}
		else if (index == 4)
		{
			UIManager.Instance.OpenWindow(GameUIList.tip);
		}
		else if (index == 5)
		{
			UIManager.Instance.OpenWindow(GameUIList.dialog);
		}
	}

	private void OnClick()
	{
		UIManager.Instance.CloseWindow(this);
	}

}

File: uLab/Assets/Lua/UI/ShopWindow.lua

local gameObject
local transform

local ShopWindow = class("ShopWindow",UIBase)
--local this = ShopWindow

local buttonNames = { "bag", "Shop", "shop", "tip", "dialog"}


function ShopWindow:OnInit(obj)
	gameObject = obj
	transform = obj.transform

	for i = 1, 5 do
		local btn = Util.FindGameObject(gameObject, 'buttons/button'..tostring(i))
		local label = Util.FindComponent(btn, 'Text', "Text")
		label.text = buttonNames[i]
		UIEventListener.SetOnClick(btn, onBtnClick)
	end

	local back = Util.FindGameObject(gameObject, "buttons/back")
	UIEventListener.SetOnClick(back, onBackClick)

end

function ShopWindow:OnEnter()
	
end

function ShopWindow:OnExit()
	
end

function ShopWindow:OnPause()
	
end

function ShopWindow:OnResume()
	
end

function onBtnClick(go)
	local index = tonumber(string.sub(go.name, string.len(go.name)))
	
	if index == 1 then
		gUIManager:OpenWindow(GameUI.bag)
	elseif index == 2 then
		gUIManager:OpenWindow(GameUI.role)
	elseif index == 3 then
		gUIManager:OpenWindow(GameUI.shop)
	elseif index == 4 then
		gUIManager:OpenWindow(GameUI.tip)
	elseif index == 5 then
		gUIManager:OpenWindow(GameUI.dialog)
	end
end

function onBackClick(go)
	CloseWindow(GameUI.shop)
end

return ShopWindow